feature-libs/customer-ticketing/components/customer-ticketing-create/customer-ticketing-create.component.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  Component,
+  ElementRef,
+  OnDestroy,
+  ViewChild,
+  ViewContainerRef,
+} from '@angular/core';
+import { LaunchDialogService, LAUNCH_CALLER } from '@spartacus/storefront';
+import { Subscription } from 'rxjs';
+import { take } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-customer-ticketing-create',
+  templateUrl: './customer-ticketing-create.component.html',
+})
+export class CustomerTicketingCreateComponent implements OnDestroy {
+  protected subscription = new Subscription();
+
+  @ViewChild('element') element: ElementRef;
+
+  constructor(
+    protected launchDialogService: LaunchDialogService,
+    protected vcr: ViewContainerRef
+  ) {}
+
+  openCreateNewTicketDialog() {
+    const dialog = this.launchDialogService.openDialog(
+      LAUNCH_CALLER.CUSTOMER_TICKETING_CREATE,
+      this.element,
+      this.vcr
+    );
+    if (dialog) {
+      this.subscription.add(dialog.pipe(take(1)).subscribe());
+    }
```

Comment:
Is OpenDialog synchronous?
Asking because you are using dialog right after

Reply:
Yes..and also this is how it is being used at various other places too.